Add noqa F821 and eventually fix

Results from ruff lint src --fix

```
src/render_engine/hookspecs.py:28:16: F821 Undefined name `Site`
src/render_engine/hookspecs.py:35:16: F821 Undefined name `Site`
src/render_engine/hookspecs.py:43:16: F821 Undefined name `Site`
src/render_engine/hookspecs.py:51:16: F821 Undefined name `page`
src/render_engine/hookspecs.py:61:16: F821 Undefined name `page`
src/render_engine/hookspecs.py:63:16: F821 Undefined name `Site`
src/render_engine/hookspecs.py:72:22: F821 Undefined name `Collection`
src/render_engine/hookspecs.py:80:16: F821 Undefined name `Site`
```

You can noqa the entire file by adding `ruff: noqa: F821`

This should be fixed long term but may require some refactor for a future version.

We can also noqa this issue in the pyproject.toml config for ruff - Again this should be refactored to no longer having this issue.

--- a/src/render_engine/collection.py
+++ b/src/render_engine/collection.py
@@ -142,7 +142,8 @@ class Collection(BaseObject):
 
         if not getattr(self, "has_archive", False):
             logging.warning(
-                "`has_archive` is set to `False` for %s. While an archive will be generated. The file will not be saved.",
+                "`has_archive` is set to `False` for %s. While an archive will be generated. \
+                The file will not be saved.",
                 self._title,
             )
             yield from ()

--- a/src/render_engine/engine.py
+++ b/src/render_engine/engine.py
@@ -75,7 +75,7 @@ def url_for(env: Environment, value: str, page: int = 0) -> str:
     routes = env.globals.get("routes")
     route = value.split(".", maxsplit=1)
 
-    if len(route) == 2 and type(route) == list:
+    if len(route) == 2 and isinstance(route, list):
         collection, route = route
         if collection := routes.get(collection, None):
             for page in collection:

--- a/src/render_engine/hookspecs.py
+++ b/src/render_engine/hookspecs.py
@@ -1,3 +1,5 @@
+# ruff: noqa: F821
+
 import typing
 
 import pluggy

--- a/src/render_engine/parsers/base_parsers.py
+++ b/src/render_engine/parsers/base_parsers.py
@@ -1,3 +1,5 @@
+# ruff: noqa: F821
+
 import pathlib
 
 import frontmatter

--- a/src/render_engine/parsers/markdown/__init__.py
+++ b/src/render_engine/parsers/markdown/__init__.py
@@ -1,3 +1,5 @@
+# ruff: noqa: F821
+
 from markdown2 import markdown
 
 from ..base_parsers import BasePageParser

